controllers/controllers/services/instances/managed/controller.go

```diff
@@ -129,6 +117,18 @@ func (r *Reconciler) ReconcileResource(ctx context.Context, serviceInstance *kor
 		return ctrl.Result{}, fmt.Errorf("failed to create client for broker %q: %w", serviceBroker.Name, err)
 	}
 
+	if !serviceInstance.GetDeletionTimestamp().IsZero() {
```

Comment:
I think we should keep the deleted/ready/failed check right after setting the observed generation to the status.

Provided that the instance is being deleted, or has reached its terminal state, we should not bother getting the plan at all.

Reply:
Since, we implemented deprovisioning functionality the finalizeCFServiceInstance method needs the service plan, offering and etc. because we call the Deprovision method in it and we call the finalizeCFServiceInstance method only in the deleted/ready/failed check

Reply:
Oh, I see. Makes sense